src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,369 @@
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Actions {
+
+static constexpr size_t kActionNameMaxSize       = 128u;
+static constexpr size_t kEndpointListNameMaxSize = 128u;
+
+static constexpr size_t kEndpointListMaxSize = 256u;
+
+class Delegate;
+
+struct ActionStructStorage : public Structs::ActionStruct::Type
+{
+    ActionStructStorage(){};
+
+    ActionStructStorage(uint16_t aAction, const CharSpan & aActionName, ActionTypeEnum aActionType, uint16_t aEpListID,
+                        BitMask<CommandBits> aCommands, ActionStateEnum aActionState)
+    {
+        Set(aAction, aActionName, aActionType, aEpListID, aCommands, aActionState);
+    }
+
+    ActionStructStorage(const ActionStructStorage & aAction) { *this = aAction; }
+
+    ActionStructStorage & operator=(const ActionStructStorage & aAction)
+    {
+        Set(aAction.actionID, aAction.name, aAction.type, aAction.endpointListID, aAction.supportedCommands, aAction.state);
+        return *this;
+    }
+
+    void Set(uint16_t aAction, const CharSpan & aActionName, ActionTypeEnum aActionType, uint16_t aEpListID,
+             BitMask<CommandBits> aCommands, ActionStateEnum aActionState)
+    {
+        actionID          = aAction;
+        type              = aActionType;
+        endpointListID    = aEpListID;
+        supportedCommands = aCommands;
+        state             = aActionState;
+        MutableCharSpan mActionName(mBuffer);
+        CopyCharSpanToMutableCharSpanWithTruncation(aActionName, mActionName);
+        name = mActionName;
+    }
+
+private:
+    char mBuffer[kActionNameMaxSize];
+};
+
+struct EndpointListStorage : public Structs::EndpointListStruct::Type
+{
+    EndpointListStorage(){};
+
+    EndpointListStorage(uint16_t aEpListId, const CharSpan & aEpListName, EndpointListTypeEnum aEpListType,
+                        const DataModel::List<const EndpointId> & aEndpointList)
+    {
+        Set(aEpListId, aEpListName, aEpListType, aEndpointList);
+    }
+
+    EndpointListStorage(const EndpointListStorage & aEpList) { *this = aEpList; }
+
+    EndpointListStorage & operator=(const EndpointListStorage & aEpList)
+    {
+        Set(aEpList.endpointListID, aEpList.name, aEpList.type, aEpList.endpoints);
+        return *this;
+    }
+
+    void Set(uint16_t aEpListId, const CharSpan & aEpListName, EndpointListTypeEnum aEpListType,
+             const DataModel::List<const EndpointId> & aEndpointList)
+    {
+        endpointListID    = aEpListId;
+        type              = aEpListType;
+        size_t epListSize = std::min(aEndpointList.size(), ArraySize(mEpList));
+
+        for (size_t index = 0; index < epListSize; index++)
+        {
+            mEpList[index] = aEndpointList[index];
+        }
+        endpoints                   = DataModel::List<const EndpointId>(Span(mEpList, epListSize));
+        MutableCharSpan mEpListName = Span(mBuffer, sizeof(mBuffer));
```

Comment:
Use MutableCharSpan on the right hand side

Reply:
Used `MutableCharSpan mEpListName(mBuffer)` instead.